src/dao/AlarmDAO.ts

```diff
@@ -438,6 +441,22 @@ export class AlarmDAO extends AbstractDAO<number, OnmsAlarm> {
    * @hidden
    */
   protected async getOptions(filter?: Filter): Promise<OnmsHTTPOptions> {
+    if (filter) {
+      this.visitFilter(filter, {
+        onRestriction: (restriction: Restriction) => {
+          if (restriction.attribute === 'isAcknowledged') {
+            let value = String(restriction.value).toLowerCase() === 'true';
```

Comment:
Is it safe to assume that the value is non null?

Reply:
`String(restriction.value)` should coerce to the string 'null' in that case which should not be truthy.

<img width="490" alt="null value" src="https://user-images.githubusercontent.com/89252/55088365-01579b80-5082-11e9-97da-da477ba647c7.png">